### PR TITLE
MM-36598 : Fix emoji search with emoji skin tone

### PR DESCRIPTION
**Pull request by jbattistispiria**
#### Summary
Fix two behaviors in the emoji search results:
-Search results for emoji now contain only emoji with user skin tone.
-Recent emoji in search result are updated when changing current user skin tone. (Before, emoji with other skin tone where added.)

#### Ticket Link
Fix https://github.com/mattermost/mattermost-server/issues/18047
Jira: https://mattermost.atlassian.net/browse/MM-36598

#### Screenshots
Search results updated with only user skin tone emoji
|  before  |  after  |
|----|----|
| image | image |

Switch user skin tone search results
|  before  |  after  |
|----|----|
| image | image |

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by koox00 — APPROVED** [review]

**Review by koox00 — CHANGES_REQUESTED** [review]
Hi @jbattistispiria thank you for the PR 💯 
I just noticed in the screenshots that custom emojis are not returned after the fix,
is that correct?

*most likely in the dev server you don't have the emojis shown in the "before" screenshot 😅 
I'm blocking the PR just to be sure, feel free to re-request once answered so I can approve!

**Comment by jbattistispiria** [comment]
@koox00 Yes, you're right. It's a screenshot from my dev server. I updated the screenshot with the custom gif added on my side.

**Review by koox00 — APPROVED** [review]
LGTM! thanks!!

File: components/emoji_picker/utils/index.ts
Comment (hmhealey) [inline components/emoji_picker/utils/index.ts:105]: Some of these changes are going to conflict with https://github.com/mattermost/mattermost-webapp/pull/10321 since I moved some of this around and made some similar changes as well. I think that's fine though. I'll wait to merge my PR until after this one is in

File: components/emoji_picker/utils/index.ts
Comment (hmhealey) [inline components/emoji_picker/utils/index.ts:76]: Similarly, this shouldn't be needed any more once that PR is in.

**Review by hmhealey — APPROVED** [review]
Sorry for the delay in this review. It looks good to me

**Comment by jgilliam17** [comment]
Thanks @jbattistispiria 
Tested, looks good to merge.
- Verified emoji search respects selected emoji skin tone.
- [E2E report](https://mattermost-cypress-report.s3.amazonaws.com/198067-e14d9b1-onprem-ent-webapp-pr-10307-mattermost/mm-ee-teste14d9b1/mochawesome.html) looks good, no PR related failures.

**Review by jgilliam17 — APPROVED** [review]

**Pull request merged**
